BlackBox/BlackBox/smallDFA.ino

```diff
@@ -0,0 +1,56 @@
+// States for Small DFA
+State A = State(InStateAction);
+State B = State(InStateAction);
+State C = State(InStateAction);
+
+void RunSmallDFA(char input[]) {
+  bool stringComplete = false;
+  FSM DFA = FSM(A);
+  DFA.update();
+
+  for (int i = 0; i < strlen(input); i++) {
+    if (input[i] == '1') {
+      if (DFA.isInState(A)) {
+        DFA.transitionTo(B);
+      }
+      if (DFA.isInState(B)) {
+        DFA.transitionTo(C);
+      }
+      if (DFA.isInState(C)) {
+        DFA.transitionTo(A);
+      }
+    } else if (input[i] == '0') {
+      if (DFA.isInState(A)) {
+        DFA.transitionTo(A);
+      }
+      if (DFA.isInState(B)) {
+        DFA.transitionTo(A);
+      }
+      if (DFA.isInState(C)) {
+        DFA.transitionTo(A);
+      }
+    }
+    // If char of input is not 1 | 0 we break and check if we are in final
+    // state.
+    else {
+      stringComplete = false;
+      Serial.println("Failed because it enters the trash state");
+      break;
+    };
+    // If we are at end of string
+    if (i == strlen(input) - 1) {
+      stringComplete = true;
+    };
+    // Moves state in dfa
+    DFA.update();
+  }
+
+  // if we finish in a final state report back that we have passed
+  if (stringComplete && DFA.isInState(C)) {
+    uint8_t reply[15] = "Trace Accepted!";
+    TransmitVerdict(reply);
+  } else {
+    uint8_t reply[13] = "Trace Failed!";
+    TransmitVerdict(reply);
+  }
+}
```

Comment:
smallDFA har ikke to kald til at transition og update som de to andre DFAs har her til sidst. Is this correct?

Reply:
No idea, men det ser ud til, at det blev brugt som en form for reset af modellen, så jeg tilføjer det. Ellers spørg @WassawRoki og @selectionator

Reply:
De siger at det skal være der

Reply:
Alright, det er tilføjet.